code/modules/surgery/organs/blood.dm

```diff
@@ -117,264 +73,218 @@
 				death()
 
 		//Bleeding out
-		var/blood_max = 0
+		var/temp_bleed = 0
 		for(var/obj/item/organ/external/temp in bodyparts)
 			if(!(temp.status & ORGAN_BLEEDING) || temp.status & ORGAN_ROBOT)
 				continue
 			for(var/datum/wound/W in temp.wounds)
 				if(W.bleeding())
-					blood_max += W.damage / 4
+					temp_bleed += W.damage / 4
 			if(temp.open)
-				blood_max += 2  //Yer stomach is cut open
-		drip(blood_max)
+				temp_bleed += 0.5
 
-//Makes a blood drop, leaking certain amount of blood from the mob
-/mob/living/carbon/human/proc/drip(var/amt as num)
+		bleed_rate = max(bleed_rate - 0.5, temp_bleed)//if no wounds, other bleed effects (heparin) naturally decreases
 
-	if(species && species.flags & NO_BLOOD) //TODO: Make drips come from the reagents instead.
-		return
+		if(bleed_rate && !bleedsuppress)
+			bleed(bleed_rate)
 
-	if(!amt)
-		return
-
-	var/amm = 0.1 * amt
-	var/turf/T = get_turf(src)
+//Makes a blood drop, leaking amt units of blood from the mob
+/mob/living/carbon/proc/bleed(amt)
+	if(blood_volume)
+		blood_volume = max(blood_volume - amt, 0)
+		if(isturf(src.loc)) //Blood loss still happens in locker, floor stays clean
+			if(amt >= 10)
+				add_splatter_floor(src.loc)
+			else
+				add_splatter_floor(src.loc, 1)
 
-	if(species.exotic_blood)
-		vessel.remove_reagent(species.exotic_blood,amm)
-		if(vessel.total_volume)
-			var/fraction = amm / vessel.total_volume
-			vessel.reaction(T, TOUCH, fraction)
-		return
+/mob/living/carbon/human/bleed(amt)
+	if(!(species && species.flags & NO_BLOOD))
+		..()
```

Comment:
This isn't correct and will prevent the blood volume from decreasing if they have exotic blood, which you don't want.

```
 /mob/living/carbon/human/bleed(amt)
 	if(!(species && species.flags & NO_BLOOD))
 	 	..()
 	 	if(species.exotic_blood)
			var/datum/reagent/R = chemical_reagents_list[get_blood_id()]
			if(istype(R))
				R.reaction_turf(get_turf(src), blood_volume)
```